api/v1/inferencepool_types.go

```diff
@@ -65,19 +67,24 @@ type InferencePoolSpec struct {
 	// this configuration into a Service resource.
 	//
 	// +required
-	Selector LabelSelector `json:"selector,omitempty,omitzero"`
+	//nolint:kubeapilinter // ignore kubeapilinter here as we don't want to use pointer here.
+	Selector LabelSelector `json:"selector"`
 
 	// TargetPorts defines a list of ports that are exposed by this InferencePool.
 	// Currently, the list may only include a single port definition.
+	//
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=1
 	// +listType=atomic
 	// +required
-	TargetPorts []Port `json:"targetPorts,omitempty"`
+	//nolint:kubeapilinter // ignore kubeapilinter here since the field is required and we don't want omitempty tag.
+	TargetPorts []Port `json:"targetPorts"`
```

Comment:
I think omitempty is fine here and all other fields which would allow us to remove `nolint` line.